src/coap_mbedtls.c

```diff
@@ -810,7 +810,7 @@ pki_sni_callback(void *p_info, mbedtls_ssl_context *ssl,
                               &m_context->pki_sni_entry_list[i].private_key);
 }
 
-#if defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
+#if defined(MBEDTLS_SSL_PROTO_DTLS) && defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
```

Comment:
I wonder if this callback function is specific to DTLS (i.e., it will presumably break PR #704)?

Reply:
Good point.  Removed this particular change and reworked the #idfef in `setup_server_ssl_session()`.  Code pushed.